### ISSUE TITLE
Add parallel versions of tests

### ISSUE THREAD

**Issue by MakisH** [precice/systemtests#59]
At the moment we run all the tests in serial (e.g. using the `Allrun` script). It would be good for every adapter to include also a parallel version (e.g. using the `Allrun_parallel` script).

We don't need to run everything in parallel, but we should make sure that at least every feature of every adapter is tested. The following features should be tested also in parallel (edit to add more):

- [ ] OpenFOAM: CHT
- [ ] OpenFOAM: CHT with nearest-projection mapping (depends on https://github.com/precice/openfoam-adapter/pull/46)
- [ ] OpenFOAM: FSI (needs a serial test case first #32)
- [ ] CalculiX: CHT
- [ ] CalculiX: FSI
~~CalculiX: SSI (needs a serial test case first)~~
- [ ] SU2: FSI
- [ ] FEniCS: CHT

Alternatively, we could convert all the current test cases to parallel, considering that they set a superset of features.

**Comment by MakisH** [comment]
If we add both serial and parallel cases, then #58 is relevant.

**Comment by uekerman** [comment]
By parallel testcase, do you refer to inter-solver parallelism (i.e. parallel coupling) or intra-solver parallelism (i.e. a solver is started on multiple ranks)?

I guess intra-solver parallelism, but then the remark about SSI above confuses me, as CCX does not support intra-solver parallelism. 

~~In the end, the purpose of the system tests is mainly to test the adapters and the interplay of the adapters with precice. In my opinion, we don't need to test every single feature of preCICE. For that, we have the integration tests.~~

> Alternatively, we could convert all the current test cases to parallel, considering that they set a superset of features.

This might not be a good idea (again assuming that you mean intra-solver parallelism). Parallel is not necessarily a superset. Also, we want to keep them as tutorials, right? Then, serial is better to start with.

Last remark: In the long run, the precice config should be the same for serial and parallel solvers, probably simplifying all this. I just created an issue: https://github.com/precice/precice/issues/347

**Comment by MakisH** [comment]
Indeed, I refer to intra-sover parallelism and indeed, this does not make sense for a CalculiX-only case.

Looking from the perspective of adapters development, I would be more confident if we also tested the decomposed cases. It is not about the preCICE features, but it's about the adapters getting/setting correct values. This could me easily messed up without noticing, therefore the automation.

Keep in mind that the system tests have a dual purpose: to test (changes in) preCICE in complete problems, but also to test (changes in) every adapter in complete problems.

**Comment by uekerman** [comment]
Yes, I completely agree that we should also test the adapters in parallel. Not every case, but we should cover all adapter features. 

Please ignore my comment: 
> In the end, the purpose of the system tests is mainly to test the adapters and the interplay of the adapters with precice. In my opinion, we don't need to test every single feature of preCICE. For that, we have the integration tests.

This was due to the fact that I thought you wanted to test parallel coupling schemes.
